tiledb/sm/c_api/tiledb.cc

```diff
@@ -1482,41 +1469,30 @@ capi_return_t tiledb_array_delete_fragments_v2(
   }
 
   // Allocate an array object
-  tiledb_array_t* array = new (std::nothrow) tiledb_array_t;
-  try {
-    array->array_ =
-        make_shared<tiledb::sm::Array>(HERE(), ctx->resources(), uri);
-  } catch (...) {
-    delete array;
-    array = nullptr;
-    throw api::CAPIStatusException("Failed to create array");
-  }
+  tiledb::sm::Array array(ctx->resources(), uri);
 
   // Set array open timestamps
-  array->array_->set_timestamp_start(timestamp_start);
-  array->array_->set_timestamp_end(timestamp_end);
+  array.set_timestamp_start(timestamp_start);
+  array.set_timestamp_end(timestamp_end);
 
   // Open the array for exclusive modification
-  throw_if_not_ok(array->array_->open(
+  throw_if_not_ok(array.open(
       static_cast<tiledb::sm::QueryType>(TILEDB_MODIFY_EXCLUSIVE),
       static_cast<tiledb::sm::EncryptionType>(TILEDB_NO_ENCRYPTION),
       nullptr,
       0));
 
   // Delete fragments
   try {
-    array->array_->delete_fragments(uri, timestamp_start, timestamp_end);
+    array.delete_fragments(uri, timestamp_start, timestamp_end);
   } catch (...) {
-    throw_if_not_ok(array->array_->close());
-    delete array;
-    array = nullptr;
-    throw api::CAPIStatusException("Failed to delete fragments");
+    throw_if_not_ok(array.close());
+    std::throw_with_nested(
+        api::CAPIStatusException("Failed to delete fragments"));
```

Comment:
I don't think the message in `throw_with_nested` adds any new information that an exception coming from `delete_fragments` wouldn't already have. Seems like unadorned `throw` would be better.